### PR TITLE
Dynamic channels 23

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: temba/channels/models.py [hunk 1]
```diff
@@ -657,7 +547,7 @@ def supports_media(cls, channel):
         with ChannelStructs or Channels
         """
         # Twilio only supports mms in the US and Canada
-        if channel.channel_type == cls.TYPE_TWILIO:
+        if channel.channel_type == 'T':
```
Comment (rowanseymour) [inline temba/channels/models.py]: you can move this into `has_attachment_support` for Twilio

File: temba/channels/models.py [hunk 2]
```diff
@@ -986,6 +876,9 @@ def release(self, trigger_sync=True):
             try:
                 # if channel is a new style type, deactivate it
                 channel_type.deactivate(self)
+            except TwilioRestException as e:
```
Comment (rowanseymour) [inline temba/channels/models.py]: why do we need this?

File: temba/ivr/models.py [hunk 3]
```diff
@@ -44,7 +44,7 @@ def hangup_test_call(cls, flow):
         runs = FlowRun.objects.filter(flow=flow, contact__is_test=True).exclude(connection=None)
         for run in runs:
             test_call = IVRCall.objects.filter(id=run.connection.id).first()
-            if test_call.channel.channel_type in [Channel.TYPE_TWILIO, Channel.TYPE_TWIML]:
+            if test_call.channel.channel_type in ['T', 'TW']:
```
Comment (rowanseymour) [inline temba/ivr/models.py]: hmm this isn't ideal - it's fine to have raw type codes in test files and handlers which are movin to courier soon - but we don't want them popping up all over the place. Can we encapsulate this in something like `ChannelType.close_ivr_session` ?

File: temba/channels/models.py [hunk 4]
```diff
@@ -716,9 +606,9 @@ def get_ivr_client(self):
         if not (self.is_active and self.org):
             return None
 
-        if self.channel_type == Channel.TYPE_TWILIO:
+        if self.channel_type == 'T':
```
Comment (rowanseymour) [inline temba/channels/models.py]: so this probably doesn't belong in this PR, but we should think about how to abstract IVR clients/functionality

File: temba/channels/models.py (hunk 2 above)
Comment (norkans7) [reply temba/channels/models.py]: We currently tell the use when we failed to remove the channel on Twilio side.
https://github.com/nyaruka/rapidpro/blob/master/temba/channels/views.py#L1324

File: temba/ivr/models.py (hunk 3 above)
Comment (norkans7) [reply temba/ivr/models.py]: I think that can be done in a different PR when abstracting the IVR functionality

File: temba/channels/models.py (hunk 2 above)
Comment (rowanseymour) [reply temba/channels/models.py]: Ok, just don't want this to be Twilio specific - maybe we can have Channel.release() return a boolean if an exception occurs?

Comment (norkans7) [reply temba/channels/models.py]: Okay

Comment (norkans7) [reply temba/channels/models.py]: What about removing the specific views handle for Twilio then and raise the exception in deactivate 
So the views will use https://github.com/nyaruka/rapidpro/blob/master/temba/channels/views.py#L1334

Comment (rowanseymour) [reply temba/channels/models.py]: Hmmm. So we have a couple of differences between Twilio channels and all other channel types:
  1. If we run into an exception removing a Twilio channel then we don't actually remove it - but for other channel types we log the error and continue with the removal.
  2. We show a custom error message about why the Twilio channel couldn't be removed

Regarding 1., my inclination is that it's better to remove the channel locally even if we can't remove it remotely
Regarding 2. custom error messages are useful and maybe we can extend that to all channel types?

Maybe ChannelType.deactivate can always throw a ValueError (or a new custom exception type) with a message, and then in Channel.release we catch and return this after completing channel removal?

Thoughts @nicpottier @ericnewcomer ?

Comment (rowanseymour) [reply temba/channels/models.py]: I think what I've described above is right but lets do it in another PR as this is already big enough
